### PR TITLE
Upgrade celery to 4.2

### PR DESCRIPTION
**Pull request by iamsobanjaved**
## Description
- Celery is now upgraded to 4.2.2
- Updated celery config

https://openedx.atlassian.net/browse/BOM-1991

### CODE REVIEW

File: requirements/constraints.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ social-auth-core<3.3.0
 
 # From base.in
 analytics-python<2.0         # https://github.com/edx/registrar/pull/39
-celery<4.0                   # https://github.com/edx/registrar/pull/66
+celery<4.3                   # Incremental upgrade of celery to latest version
```
Comment (jmbowman) [inline requirements/constraints.txt]: I just wrote https://openedx.atlassian.net/browse/BOM-1995 which will hopefully enable further upgrades later if we need them.

File: registrar/celery.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 app = Celery('registrar')
 
-app.config_from_object(settings)
+app.config_from_object('django.conf:settings')
```
Comment (jmbowman) [inline registrar/celery.py:12]: You'll need `app.conf.task_protocol = 1` before this line for django-user-tasks to work properly in actual deployments.

File: registrar/settings/base.py [hunk 3]
```diff
@@ -146,6 +146,11 @@
     CELERY_BROKER_VHOST
 )
 
+BROKER_TRANSPORT_OPTIONS = {
+    'fanout_patterns': True,
+    'fanout_prefix': True,
+}
```
Comment (jmbowman) [inline registrar/settings/base.py]: Hmm, this change should get deployed before the deployment that upgrades celery.  Otherwise there's a risk of losing some messages.  Put this in a separate PR and release that first?

Comment (iamsobanjaved) [reply registrar/settings/base.py]: @jmbowman `app.conf.task_protocol = 1` add this to that PR too?

Comment (jmbowman) [reply registrar/settings/base.py]: No, I don't think that option was added until 4.0, so it has to stay in this one.

**Review by awais786 — APPROVED** [review]

**Review by jmbowman — APPROVED** [review]
Looks good, just make sure the other PR with the broker settings changes has been deployed before merging this.

**Comment by iamsobanjaved** [comment]
> Looks good, just make sure the other PR with the broker settings changes has been deployed before merging this.

@jmbowman that is deployed to the stage as per newrelic but not on prod
